### ISSUE TITLE
Hello , Is this package able to train a multi-label dataset with one-hot encoding?

### ISSUE THREAD

**Issue by Mahiro2211** [KevinMusgrave/pytorch-metric-learning#669]
It is a great package that improves my efficiency , When I test cifar10, for the one-hot label , I can use
```python 
label = torch.argmax(label,dim=1)
```
to transform one-hot label but When I test it on some one-hot label I can't find a nice method to deal with a multi-label dataset.

at first, I saw this [issue](https://github.com/KevinMusgrave/pytorch-metric-learning/issues/178#issuecomment-675611566) it tells me a way to put in multi-label, but I want to further custom it because I need to construct a similarity matrix
```python
label =  torch.matmul(label,label.t())
# For multi-label dataset , if there is one label shared by two samples I mark it as the same
```

I hope to receive a response from you soon. Thank you.

**Comment by KevinMusgrave** [comment]
Unfortunately there currently isn't a way to pass in a custom label comparison function into miners or loss functions. It would be a good idea to add this feature though, so I will keep this issue open.

**Comment by Mahiro2211** [comment]
Thank you very much for your response. I will try it

**Comment by Mahiro2211** [comment]
Thank you for your response.I will try it
